turbo/snapshotsync/freezeblocks/block_snapshots.go

```diff
@@ -1434,27 +1444,45 @@ func hasIdxFile(sn snaptype.FileInfo, logger log.Logger) bool {
 	dir := sn.Dir()
 	fName := snaptype.IdxFileName(sn.Version, sn.From, sn.To, sn.Type.String())
 	var result = true
+
+	segment, err := seg.NewDecompressor(sn.Path)
+
+	if err != nil {
+		return false
+	}
+
+	defer segment.Close()
+
 	switch sn.Type.Enum() {
 	case snaptype.Enums.Headers, snaptype.Enums.Bodies, snaptype.Enums.BorEvents, snaptype.Enums.BorSpans, snaptype.Enums.BeaconBlocks:
 		idx, err := recsplit.OpenIndex(filepath.Join(dir, fName))
 		if err != nil {
 			return false
 		}
-		idx.Close()
+		defer idx.Close()
+
+		return idx.ModTime().After(segment.ModTime())
```

Comment:
let's move "modtime" check feature to another PR

Reply:
ok - TBH I though that this was a reversion I made - as I thought this mod check was already in place.

I agree that we should add this as a file - but we need the check - what happens otherwise is that you get a panic - or worse unexpected misses because the index does not match the file.